packages/core/test/app/ovm/deciders/hash-preimage-existence-decider.spec.ts

```diff
@@ -0,0 +1,133 @@
+import '../../../setup'
+
+import MemDown from 'memdown'
+import {
+  CannotDecideError,
+  HashPreimageExistenceDecider,
+} from '../../../../src/app/ovm/deciders'
+import { BaseDB } from '../../../../src/app/db'
+import { Md5Hash } from '../../../../src/app/utils'
+import {
+  Decider,
+  Decision,
+  ImplicationProofElement,
+} from '../../../../src/types/ovm/decider.interface'
+import * as assert from 'assert'
+import { DB } from '../../../../src/types/db'
+
+/*********
+ * TESTS *
+ *********/
+
+describe('HashPreimageExistenceDecider', () => {
+  const preimage: Buffer = Buffer.from('really great preimage')
+  const hash: Buffer = Md5Hash(preimage)
+
+  describe('Constructor', () => {
+    it('should initialize', async () => {
+      new HashPreimageExistenceDecider(
+        new BaseDB(new MemDown('') as any, 256),
+        Md5Hash
+      )
+    })
+  })
+
+  describe('decide', () => {
+    let decider: Decider
+    let db: DB
+    let memdown: any
+
+    beforeEach(() => {
+      memdown = new MemDown('')
+      db = new BaseDB(memdown, 256)
+      decider = new HashPreimageExistenceDecider(db, Md5Hash)
+    })
+
+    afterEach(async () => {
+      await db.close()
+      memdown = undefined
+    })
+
+    it('should decide true for valid preimage', async () => {
+      const decision: Decision = await decider.decide({ hash }, { preimage })
+
+      decision.outcome.should.equal(true)
+      decision.justification.length.should.equal(1)
+
+      const justification: ImplicationProofElement = decision.justification[0]
+      justification.implication.decider.should.equal(decider)
+      justification.implication.input['hash'].should.equal(hash)
+      justification.implicationWitness['preimage'].should.equal(preimage)
+    })
+
+    it('should throw invalid preimage', async () => {
+      try {
+        await decider.decide({ hash }, { preimage: Buffer.from('womp womp') })
+      } catch (e) {
+        if (e instanceof CannotDecideError) {
+          assert(true, 'This is expected to happen')
+        } else {
+          assert(false, 'CannotDecideError was expected.')
+        }
```

Comment:
I think you may be able to use https://www.chaijs.com/api/bdd/#method_throw instead of the `try {} catch {}`. That said, I personally don't remember using `throw()` with an async function so it could be a little more tricky.

That said, I do like the 'low tech-ness' of your current approach so take this as just an alternative not so much a suggestion.